src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,97 @@
+        switch (datePart) {
+            case DAY:
+                return String.valueOf(LocalDateTime.now().getDayOfMonth());
+            case YEAR:
+                return String.valueOf(LocalDateTime.now().getYear());
+            case MONTH:
+                return String.valueOf(LocalDateTime.now().getMonth());
+            case FULL:
+                return String.valueOf(LocalDate.now());
+            default:
+                throw new DateTimeException("Can't return any date!");
+        }
     }
 
-    /**
-     * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
-     * Return Optional of a date built from these elements.
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (RuntimeException e) {
+            return Optional.empty();
+        }
     }
 
-    /**
-     * Given the time and the number of hours to add, return the changed time.
-     */
     public LocalTime addHours(LocalTime localTime, Integer hoursToAdd) {
-        return LocalTime.now();
+        return localTime.plusHours(hoursToAdd);
     }
 
-    /**
-     * Given the time and the number of minutes to add, return the changed time.
-     */
     public LocalTime addMinutes(LocalTime localTime, Integer minutesToAdd) {
-        return LocalTime.now();
+        return localTime.plusMinutes(minutesToAdd);
     }
 
-    /**
-     * Given the time and the number of seconds to add, return the changed time.
-     */
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
 
-    /**
-     * Given the date and the number of weeks to add, return the changed date.
-     */
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
-    /**
-     * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
-     */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        return someDate.isAfter(LocalDate.now())
+                ? someDate + " is after " + LocalDate.now()
+                : someDate.isBefore(LocalDate.now())
+                ? someDate + " is before " + LocalDate.now()
+                : someDate + " is today";
```

Comment:
Common mistakes
Avoid calling LocalDate.now multiple times in one method. Let’s create a variable for this purpose.